app/src/main/java/com/cornellappdev/android/eatery/ExpandableListAdapter.java

```diff
@@ -81,15 +94,58 @@ public View getGroupView(int i, boolean b, View view, ViewGroup viewGroup) {
             view = infalInflater.inflate(R.layout.list_view_header, viewGroup, false);
         }
 
-        CafeteriaModel m = (CafeteriaModel) getGroup(i);
+        String m = (String) getGroup(i);
         TextView headerText = view.findViewById(R.id.header);
-        headerText.setText(m.getNickName());
-
-        // TODO(lesley): Add and fix isOpen() -- mb use color or sort for open eateries to bubble
-        // to the top
-        //TextView timetext = view.findViewById(R.id.time);
-        //timetext.setText(m.isOpen());
-
+        headerText.setText(m);
+
+        // TODO(lesley): Not all the times show up for the meals, esp Risley + Okenshields
+        TextView timetext1 = view.findViewById(R.id.time1);
+
+        try {
+            CafeteriaModel myCafe = null;
+            int count = 0;
+            while(count < cafeData.size())
+            {
+                if(m.equals(cafeData.get(count).getNickName()))
+                {
+                    myCafe = cafeData.get(count);
+                }
+                count++;
+            }
+
+            ArrayList<MealModel> day = myCafe.getWeeklyMenu().get(dateOffset);
+            int length = day.size() - 1;
+            MealModel meal = day.get(mealIndex - (2-length));
+
+            SimpleDateFormat localDateFormat = new SimpleDateFormat("h:mm a");
+            String localEndTime = localDateFormat.format(meal.getEnd());
+            String localStartTime = localDateFormat.format(meal.getStart());
+
+            Date date = new Date();
+            SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+            dateFormat.setTimeZone(TimeZone.getTimeZone("America/New_York"));
+            String dateStartTime = dateFormat.format(meal.getStart());
+
+            String dateCurrent = dateFormat.format(date);
+
+            int t1 = Integer.parseInt(dateCurrent.substring(dateCurrent.length() - 2,
+                    dateCurrent.length()));
+            int t2 = Integer.parseInt(dateStartTime.substring(dateStartTime.length()
+                    - 2, dateStartTime.length()));
+
+            //meal date seems to be off by one day
+            if(myCafe.getCurrentStatus()==CafeteriaModel.Status.OPEN
+                    && t1 == t2+1 ) {
+                timetext1.setText("Open from " + localStartTime + " to " + localEndTime);
```

Comment:
Remember to change color of timetext based on open/closed + set typeface to "san-serif-medium"